tests/main/refresh-app-awareness/task.yaml

```diff
@@ -0,0 +1,34 @@
+summary: Ensure that foreground applications block app refresh.
+details: |
+    When the refresh-app-awareness feature is enabled running snap processes,
+    mainly foreground applications, will block the refresh of said snap.
+prepare: |
+    # This feature depends on the release-app-awareness feature
+    snap set core experimental.refresh-app-awareness=true
+    snap pack test-snapd-refresh.v1
+    snap pack test-snapd-refresh.v2
+restore: |
+    snap remove test-snapd-refresh
+    rm -f test-snapd-refresh-{1,2}_all.snap
+    rmdir /sys/fs/cgroup/pids/snap.test-snapd-refresh.sh || true
+    rmdir /sys/fs/cgroup/pids/snap.test-snapd-refresh.version || true
+    # TODO: There is currently no way to unset configuration options.
+    # Once this is fixed please uncomment this line:
+    # snap unset core experimental.refresh-app-awareness
+execute: |
+    # Install v1 and see that it runs as expected.
+    snap install --dangerous test-snapd-refresh_1_all.snap
+    test-snapd-refresh.version | MATCH v1
+    # Run a sleeper app to keep the snap busy
+    test-snapd-refresh.sh -c 'exec sleep 1h' &
+    pid=$!
+    trap 'kill '"$pid"' || true' EXIT
```

Comment:
why not `trap "kill $pid || true" EXIT`?

Reply:
Because then shellcheck says you gotta quote "$pid" inside the already quoted trap command argument I and don't like escaping double quotes :)

Reply:
`trap "kill '$pid' || true" EXIT` would still read better than your current one tho

Reply:
It would require a shellcheck directive though.